pykern/pkio.py

```diff
@@ -125,6 +125,30 @@ def has_file_extension(filename, to_check):
     return e in to_check
 
 
+def is_binary(filename, threshold=0.30):
+    """Uses heuristics to guess whether file is binary data
+
+    Args:
+        filename (str|py.path.local): file to check
+        threshold (float): allowed percentage of assumed non-textual characters
+
+    Returns:
+        bool: True if file is likely binary data
+    """
+    if threshold >= 1:
+        raise AssertionError(
+            f"threshold={threshold} must be < 1 (represents percentage)"
+        )
+    with open(filename, "rb") as f:
+        b = f.read(512)
+        if not b:
```

Comment:
close the file at this point, that is,

```py
with open() as f
    b = f.read()
if not b:
```

The conditional on `b` should be robust, e.g. `if (l := len(b)) <= 0`. `l` is used below so might as well do the check around that.